### PR TITLE
Added Digitizing ex.  

### PR DESCRIPTION
**Pull request by Santhosh-M31**

### CODE REVIEW

File: introduction-to-qgis.Rmd [hunk 1]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-15. 
+15. To start digitizing, select the ``banglore_lakes`` layer. Click the *Toggle editing* and the *Add Polygon Feature* button. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_15.png')
 ```
 
-16. 
+16. Enable snap setting by clicking *Enable Snapping* button. Zoom to any water body region. Starting from an edge, keep on adding vertices using left-click. Once the polygon is fully covered, right-click to complete the feature digitization.
```
Comment (spatialthoughts) [inline introduction-to-qgis.Rmd:1050]: Zoom to any water body region -> zoom to any part of the map containing a waterbody.

File: introduction-to-qgis.Rmd [hunk 2]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-16. 
+16. Enable snap setting by clicking *Enable Snapping* button. Zoom to any water body region. Starting from an edge, keep on adding vertices using left-click. Once the polygon is fully covered, right-click to complete the feature digitization. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_16.png')
 ```
 
-17. 
+17. Now the *Feature Attributes* dialog will popup to get the attributes value. By referring to the ``OSM Standard`` basemap, enter the *name* and select the *status* of the waterbody. Click *Ok*.
```
Comment (spatialthoughts) [inline introduction-to-qgis.Rmd:1056]: Now the *Feature Attributes* dialog will popup to get the attributes value -> Now you will be prompted by the *Feature Attributes* dialog to enteer the attributes values for the feature you just created.

File: introduction-to-qgis.Rmd [hunk 3]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-17. 
+17. Now the *Feature Attributes* dialog will popup to get the attributes value. By referring to the ``OSM Standard`` basemap, enter the *name* and select the *status* of the waterbody. Click *Ok*. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_17.png')
 ```
 
-18. 
+18. Now the polygons show the fully digitized feature.
```
Comment (spatialthoughts) [inline introduction-to-qgis.Rmd:1062]: polygons -> polygon

File: introduction-to-qgis.Rmd [hunk 4]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
+23. Now let's save the project, click *Project &rarr; Save*. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_23.png')
 ```
 
 ### 4.1.1 Challenge
 
-1. 
+1. Style the layer based on the *status* column. This column has categorical values. We can use this to differentiate the different statuses of each waterbody.
```
Comment (spatialthoughts) [inline introduction-to-qgis.Rmd:1100]: statuses -> status

**Review by spatialthoughts — CHANGES_REQUESTED** [review]

File: introduction-to-qgis.Rmd [hunk 5]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
 ## 4.1 Digitize Historic Lakes
 
-We will take the georeferenced map created in the previous exercise and create a polygon layer representing the lakes in the city of Bengaluru in 1924. Bengaluru has experienced rapid urbanization and many of these historic lakes no longer exist. This exercise will allow you to create a vector layer and assign attributes to them whether they are healthy, lost or partially lost.
+In this exercise, you will digitize all water bodies in the city of Banglore, India. This city has experienced urbanization at a rapid phase, and due to this development, some waterbodies have been compromised as urban buildings. We will use the georeferenced scanned map from the previous exercise to map all the water bodies in 1924 and label their current status. Finally, we will create a vector layer and assign attributes to them, whether healthy, lost, or partially lost.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:956]: Updated.

File: introduction-to-qgis.Rmd [hunk 6]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-1. 
+1. Drag and drop the ``Digitizing.qgz`` project file to the canvas. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_01.png')
 ```
 
-2. 
+2. This project contains the ``OSM Standard`` basemap and the ``Banglore_1924_modified`` georeferenced scanned map. To digitize the waterbodies, let us create a new vector layer. Click on the *New GeoPackage Layer...* icon from the toolbar.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:964]: Updated.

File: introduction-to-qgis.Rmd [hunk 7]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
-3. 
+
+3. In the *New GeoPackage Layer* dialog, click on the *...* next to *Database* and browse to the project location. Enter the file name as ``banglore_lakes`` and click *Ok*. Now the *Table name* will be auto-populated as ``banglore_lakes``. Choose ``MultiPolygon`` as *Geometry type*. Let the CRS be in default ``EPSG:4326`` projection.  
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_03.png')
 ```
-4. 
+
+4. In the new layer, let us add some basic fields. First, let us add the name field. Under the *New Field* section, in *Name* enter ``name``, in *Type* choose ``Text Data``, and in *Maximum length*, enter ``50``. Now click *Add to Fields List* button.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:976]: Updated.

File: introduction-to-qgis.Rmd [hunk 8]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
-7. 
+
+7. Let us inspect the attribute table of the new layer. Right-click on the ``banglore_lakes`` layer and click *Open Attribute Table*. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_07.png')
 ```
-8. 
+
+8. In the banglore_lakes attribute table, there are three fields. ``fid`` will be unique and autogenerated, ``name`` and ``status`` must be fed while digitizing the waterbodies. Close the attribute table.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1000]: Updated

File: introduction-to-qgis.Rmd [hunk 9]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
+
+10. In the *Layer Properties* dialog, choose *Attribute Form*. Under *Fields*, select *status*. Choose the *Widget Type* as ``Value Map``. Value Map allows us to create a drop-down while adding field values for each feature(polygon).
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_10.png')
 ```
 
-11. 
+
+11. Enter *Value* as ``1``, *Description* as ``Healthy``. *Value* as ``2``, *Description* as ``Partially Lost``, and *Value* as ``3``, *Description* as ``Lost``. The status field will be set as a drop-down with these three values.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1020]: Updated.

File: introduction-to-qgis.Rmd [hunk 10]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-12. 
+12. We require this field's input must be given for every feature, so let's make this a mandatory field. Under *Constraints* check *Not null* and *Enforce not null constraint*. Click *Ok* to save the changes and close the properties dialog. 
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_12.png')
 ```
 
-13. 
+13. In the QGIS main canvas, before editing, let us enable the snapping toolbar. This toolbar will help select the nearby vertex corners and avoid invalid geometries. Right-click on the toolbar and check the *Snapping Toolbar* to enable it.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1032]: Updated.

File: introduction-to-qgis.Rmd [hunk 11]
```diff
@@ -949,145 +949,155 @@ In this exercise we used the **Polynomial 2** technique. For datasets that requi
...
 
-14. 
+14. Now the snapping toolbar will be added to the main QGIS.  
 
 ```{r echo=FALSE, fig.align='center', out.width='75%'}
 knitr::include_graphics('images/introduction_to_qgis/digitizing_14.png')
 ```
 
-15. 
+15. To start digitizing, select the ``banglore_lakes`` layer. Click the *Toggle editing* and the *Add Polygon Feature* button.
```
[the remark this answers is not in the document]
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1044]: Updated.

File: introduction-to-qgis.Rmd (hunk 1 above)
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1050]: Updated.

File: introduction-to-qgis.Rmd (hunk 2 above)
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1056]: Updated.

File: introduction-to-qgis.Rmd (hunk 3 above)
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1062]: Updated.

File: introduction-to-qgis.Rmd (hunk 4 above)
Comment (Santhosh-M31) [reply introduction-to-qgis.Rmd:1100]: Updated.

**Pull request merged**